index.test.js

```diff
@@ -0,0 +1,74 @@
+const http = require('http');
+const supertest = require('supertest');
+const handler = require('./src/handler');
+const parseEnv = require('./src/parse-env');
+
+const describeIfEnv = (process.env.AIRTABLE_API_KEY && process.env.AIRTABLE_BASE_ID) ? describe : xdescribe;
+
+const getExistingPosts = async () => {
+  const config = parseEnv(process.env);
+  const server = http.createServer(handler(config));
+
+  return (await supertest(server).get('/v0/Posts')).body;
+};
+
+describeIfEnv('integration', () => {
+  this.config = parseEnv(process.env);
+
+  describe('allowedMethods "*"', () => {
+
+    beforeEach(() => {
+      this.config.allowedMethods = '*';
+      this.server = http.createServer(handler(this.config));
+    });
+
+    it('Can make GET request', async () => {
+      this.response = await supertest(this.server)
+        .get('/v0/Posts');
+
+      expect(this.response.statusCode).toBe(200);
+      expect(this.response.body).toBeTruthy();
+    });
+
+    it('Can make POST request', async () => {
+      this.response = await supertest(this.server)
+        .post('/v0/Posts')
+        .send({fields: {title: 'Test POST request'}});
+
+      expect(this.response.statusCode).toBe(200);
+      expect(this.response.body).toBeTruthy();
+    });
+
+    it('Can make PUT request', async () => {
```

Comment:
A common pattern I've seen for integration tests is to name the test suites after the endpoints they're testing. I think this is a nice pattern to follow:

```js
describe('GET /v0/Table', () => {
  // Whatever you're testing, for example:
  it('returns a successful response', () => { /* ... */ });
  it('responds with JSON', () => { /* ... */ });
  it('correctly updates fields', () => { /* ... */ });
});
```

We can wrap these in various contexts to describe them better:

```jsx
describe('when all methods are allowed', () => {
  describe('GET /v0/Posts', () => { ... });
  describe('GET /v0/Table%20With%2FSlashes%20In%20The%20Name', () => { ... });
  describe('PUT /v0/Posts', () => { ... });
  describe('PATCH /v0/Posts', () => { ... });
});

describe('when only GET requests are allowed', () => {
  describe('GET /v0/Posts', () => { ... });
  describe('PUT /v0/Posts', () => { ... });
});
```

This is just a common pattern I've seen — we don't _need_ to follow it if the structure seems like a hindrance, since we are making a proxy. But thought I'd mention it since you're already close to following this structure anyways.

Reply:
I think this seems reasonable. My only question is about the naming. I'm thinking a hierarchy like this:

```
describeIfEnv('micro-airtable-api', () => {
  ...
  describe('when `allowedMethods="*"`', () => {
    ...
    describe('GET /v0/Posts', async () => {
      ...
      it('returns successful response', () => {
        ...
      });
      it('contains a response body', () => {
        ...
      });
...
```

Reply:
Yeah, good point. The structure you shared makes sense to me.

Though, for our uses, I think we can probably get away with only just the first `it` block — I think every response other than a `HEAD` request will have a body. Or have you found otherwise?

Reply:
No, I think that's true. Agreed, it's simpler to just test the status code.